src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,498 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+        let deviceType = "disk";
+        if (['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(diskParams.storagePool.type) > -1) {
+            const volume = diskParams.storagePool.volumes.find(vol => vol.name === diskParams.existingVolumeName);
+            if (volume && volume.format) {
+                format = volume.format;
+                if (volume.format === "iso")
+                    deviceType = "cdrom";
             }
-        } else if (this.state.mode === USE_EXISTING) {
-            if (this.state.mode !== CUSTOM_PATH && !this.state.existingVolumeName)
-                validationFailed.existingVolumeName = _("Please choose a volume");
         }
+        setDiskParams(diskParams => ({
+            ...diskParams,
+            format,
+            device: deviceType,
+        }));
+    }, [diskParams.storagePool, diskParams.existingVolumeName]);
 
-        return validationFailed;
-    }
+    useEffect(() => {
+        // Initial state settings and follow up state updates after 'device' is changed
 
-    existingVolumeNameDelta(value, poolName) {
-        const { storagePools, vm } = this.state;
-        const stateDelta = { existingVolumeName: value };
-        const pool = storagePools.find(pool => pool.name === poolName && pool.connectionName === vm.connectionName);
-        if (!pool)
-            return stateDelta;
+        // According to https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms (section about 'target'),
+        // scsi is the default option for libvirt for cdrom devices
+        let newBus = diskParams.device === "cdrom" && "scsi";
 
-        stateDelta.format = getDefaultVolumeFormat(pool);
-        let deviceType = "disk";
-        if (['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(pool.type) > -1) {
-            const volume = pool.volumes.find(vol => vol.name === value);
-            if (volume && volume.format) {
-                stateDelta.format = volume.format;
-                if (volume.format === "iso")
-                    deviceType = "cdrom";
+        if (!newBus) {
+            // If disk with the same device exists, use the same bus too
+            for (const disk of Object.values(vm.disks)) {
+                if (disk.device === diskParams.device) {
+                    newBus = disk.bus;
+                    break;
+                }
             }
+            newBus = newBus || "virtio";
         }
 
-        this.onValueChanged("device", deviceType);
+        setDiskParams(diskParams => ({ ...diskParams, busType: newBus }));
+    }, [diskParams.device, vm.disks]);
 
-        return stateDelta;
-    }
+    useEffect(() => {
+        // Initial state settings and follow up state updates after 'busType' is changed
+        const existingTargets = Object.getOwnPropertyNames(vm.disks);
+        const availableTarget = isMediaInsertion ? disk.target : getNextAvailableTarget(existingTargets, diskParams.busType);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231120-122732-13cb53e1-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)